pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -0,0 +1,367 @@
+package gkenetworkparamset
+
+import (
+	"context"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/onsi/gomega"
+	"google.golang.org/api/compute/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+	"k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/fake"
+	networkinformers "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	"k8s.io/component-base/metrics/prometheus/controllers"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+type testGKENetworkParamSetController struct {
+	ctx                    context.Context
+	stop                   context.CancelFunc
+	networkClient          *fake.Clientset
+	networkInformerFactory networkinformers.SharedInformerFactory
+	clusterValues          gce.TestClusterValues
+	controller             *Controller
+	metrics                *controllers.ControllerManagerMetrics
+	cloud                  *gce.Cloud
+}
+
+func setupGKENetworkParamSetController() *testGKENetworkParamSetController {
+	fakeNetworking := fake.NewSimpleClientset()
+	networkInformerFactory := networkinformers.NewSharedInformerFactory(fakeNetworking, controller.NoResyncPeriodFunc())
+	testClusterValues := gce.DefaultTestClusterValues()
+	fakeGCE := gce.NewFakeGCECloud(testClusterValues)
+	controller := NewGKENetworkParamSetController(
+		fakeNetworking,
+		networkInformerFactory.Networking().V1alpha1().GKENetworkParamSets(),
+		networkInformerFactory.Start,
+		fakeGCE,
+	)
+	metrics := controllers.NewControllerManagerMetrics("test")
+	return &testGKENetworkParamSetController{
+		networkClient:          fakeNetworking,
+		networkInformerFactory: networkInformerFactory,
+		clusterValues:          testClusterValues,
+		controller:             controller,
+		metrics:                metrics,
+		cloud:                  fakeGCE,
+	}
+}
+
+func TestControllerRuns(t *testing.T) {
+	ctx, stop := context.WithCancel(context.Background())
+	defer stop()
+	testVals := setupGKENetworkParamSetController()
+	go testVals.controller.Run(ctx.Done(), testVals.metrics)
+}
+
+func TestAddValidParamSetSingleSecondaryRange(t *testing.T) {
```

Comment:
I didn't check the tests on details but at a first sight it seems that they can be moved to table based tests

Reply:
noting: upon futher discussion it does not make sense to table test this